src/Functions/FunctionsJSON.cpp

```diff
@@ -1332,13 +1383,69 @@ class JSONExtractRawImpl
 
     static bool insertResultToColumn(IColumn & dest, const Element & element, std::string_view)
     {
-        ColumnString & col_str = assert_cast<ColumnString &>(dest);
-        auto & chars = col_str.getChars();
-        WriteBufferFromVector<ColumnString::Chars> buf(chars, AppendModeTag());
+        if (dest.getDataType() == TypeIndex::LowCardinality)
+        {
+            ColumnString::Chars chars;
+            WriteBufferFromVector<ColumnString::Chars> buf(chars, AppendModeTag());
+            chars.push_back(0);
+            traverse(element, buf);
+            buf.finalize();
+            std::string str = reinterpret_cast<const char *>(chars.data());
+            chars.push_back(0);
+            assert_cast<ColumnLowCardinality &>(dest).insertData(str.data(), str.size());
+        }
+        else
+        {
+            ColumnString & col_str = assert_cast<ColumnString &>(dest);
+            auto & chars = col_str.getChars();
+            WriteBufferFromVector<ColumnString::Chars> buf(chars, AppendModeTag());
+            traverse(element, buf);
+            buf.finalize();
+            chars.push_back(0);
+            col_str.getOffsets().push_back(chars.size());
+        }
+        return true;
+    }
+
+    // We use insertResultToFixedStringColumn in case we are inserting raw data in a FixedString column
+    static bool insertResultToFixedStringColumn(IColumn & dest, const Element & element, std::string_view)
+    {
+        ColumnFixedString::Chars chars;
+        WriteBufferFromVector<ColumnFixedString::Chars> buf(chars, AppendModeTag());
+        traverse(element, buf);
+        buf.finalize();
+
+        auto & col_str = assert_cast<ColumnFixedString &>(dest);
+
+        if (chars.size() > col_str.getN())
+            return false;
+
+        chars.push_back(0);
+        std::string str = reinterpret_cast<const char *>(chars.data());
+
+        auto padded_str = str + std::string(col_str.getN() - std::min(col_str.getN(), str.length()), '\0');
+        col_str.insertData(str.data(), str.size());
```

Comment:
We can use `WriteBufferFromOwnString` here instead of `Chars`, creating `str` and `padded_str`. After writing to buffer is done, we can use `resize(col_str.getN(), '\0')`. Or ypu can leave `Chars` and use only it. By now it seems like creating to new strings `str` and `padded_str` is not necessary

Reply:
I have a doubt about this case. Should I do the check `(chars.size() > col_str.getN())` after or before pushing the `0`?
I am asking because if I change it to use just chars, if I do the check before the push, I am getting the following exception

`DB::Exception: Too large string for FixedString column: While processing JSONExtract('{"a": 123456}', 'a', 'FixedString(6)'). (TOO_LARGE_STRING_SIZE)`

If I do it the push before the check, then a case like `SELECT JSONExtract('{"a": 123456}', 'a', 'FixedString(6)')` returns empty, as the size of the string plus the 0 is bigger than N (6), but it seems wrong.

Which case would be corret?

Reply:
I just realized that both are not correct. In `FixedString` we can store data without `\0` at the end. This rule applies only for `String` column. So we don't need to do `push_back('\0')` when we work with FixedString

Reply:
All easier then. Changed